QR_extraction_service.ipynb

```diff
@@ -74,12 +74,23 @@
       "cell_type": "code",
       "source": [
         "import json\n",
-        "from decimal import Decimal\n",
         "from fhir.resources.observation import Observation\n",
         "from fhir.resources.codeableconcept import CodeableConcept\n",
         "from fhir.resources.coding import Coding\n",
         "\n",
-        "def create_observation(link_id, text, answer):\n",
+        "def extract_units(questionnaire_file):\n",
+        "    units_map = {}\n",
+        "    with open(questionnaire_file, 'r') as f:\n",
+        "        questionnaire = json.load(f)\n",
+        "        for item in questionnaire['item']:\n",
+        "            link_id = item['linkId']\n",
+        "            if 'extension' in item:\n",
+        "                for extension in item['extension']:\n",
+        "                    if 'url' in extension and extension['url'] == 'http://hl7.org/fhir/StructureDefinition/questionnaire-unit':\n",
+        "                        units_map[link_id] = extension['valueCoding']['display']\n",
+        "    return units_map\n",
+        "\n",
+        "def create_observation(link_id, text, answer, units=None):\n",
```

Comment:
we probably don't want to hard code loinc since not everything is a loinc codes. Two thoughts here:
1. Find a question on loinc that represents the question you're asking (e.g. some questions about chief complaint) and plug that into your questionnaire so using loinc would be appropriate. I'd suggest this, but quite low on the priority. 
2. Create a lookup that will check if the question is loinc before adding it (this is likely impractical and very difficult right now).